### ISSUE TITLE
Add reflected spectrum encoder

### ISSUE THREAD

**Issue by wsnoble** [Noble-Lab/casanovo#214]
Try adding a second spectrum with (m_pre - m_i)/z_i as the m/z values, a la the Panda paper.

**Comment by bittremieux** [comment]
Should we provide a second spectrum as input, or just add the reflected peaks to the original input spectrum?

**Comment by wsnoble** [comment]
No, we should start by mimicking what is done in Pandanovo, which is to create a separate encoder for the reflected spectrum. It may be worth also trying the simpler approach subsequently.

**Comment by GauravSRC** [comment]
Hi @bittremieux and @wsnoble , I’m interested in working on this issue. Just wanted to quickly check - is this still open, or has it already been addressed in another branch  but not yet merged into main? Thanks!

**Comment by wsnoble** [comment]
I think @Lilferrit has implemented this.  @Lilferrit , can you confirm?

**Comment by Lilferrit** [comment]
I have indeed implemented this. It provided a small boost in performance but nothing significant.

<img width="1333" height="750" alt="Image" src="https://github.com/user-attachments/assets/3cefe95c-164d-43e5-a6e1-eda8e499e929" />
